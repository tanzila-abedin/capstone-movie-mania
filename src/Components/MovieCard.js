import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const MovieCard = ({ movie }) => (
  <div className="col-md-3 mb-5">
    <div className="card card-body bg-dark text-center h-100">
      <img className="w-100 mb-2" src={movie.image} alt="Movie Cover" />
      <h5 className="text-light card-title">
        {movie.title}
        {' '}
        -
        {' '}
        {movie.year}
      </h5>
      <Link className="btn btn-primary" to={`/singleMovie/${movie.id}`}>
        Details
      </Link>
    </div>
  </div>
);

MovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.string,
    rank: PropTypes.string,
    rankUpDown: PropTypes.string,
    title: PropTypes.string,
    fullTitle: PropTypes.string,
    year: PropTypes.string,
    image: PropTypes.string.isRequired,
    crew: PropTypes.string,
    imDbRating: PropTypes.string,
    imDbRatingCount: PropTypes.string,
  }).isRequired,
};

export default MovieCard;
