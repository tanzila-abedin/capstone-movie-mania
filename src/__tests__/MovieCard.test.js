import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter, MemoryRouter as Router } from 'react-router-dom';
import MovieCard from '../Components/MovieCard';

test('MovieCard snapshot with react and redux  ', () => {
  const image = { image: 'empty' };
  const app = renderer.create(
    <BrowserRouter>
      <MovieCard movie={image} />
    </BrowserRouter>,
  ).toJSON();
  expect(app).toMatchSnapshot();
});

test('MovieCardcomponent is defined', () => {
  const image = { image: 'empty' };
  const component = render(
    <BrowserRouter>
      <MovieCard movie={image} />
    </BrowserRouter>,
  );
  expect(component).toBeDefined();
});

test('For presence of classname in the div container', async () => {
  const image = { image: 'empty' };
  const component = render(
    <Router>
      <MovieCard movie={image} />
    </Router>,
  );
  const movieCard = await component.findByTestId('movieCard');
  expect(movieCard.className).toContain('card-body', 'opacity');
});
