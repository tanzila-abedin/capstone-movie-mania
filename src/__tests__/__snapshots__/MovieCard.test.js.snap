// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard snapshot with react and redux   1`] = `
<div
  className="col-md-3 mb-5"
>
  <div
    className="card card-body opacity text-center h-100"
    data-testid="movieCard"
  >
    <img
      alt="Movie Cover"
      className="w-100 mb-2"
      src="empty"
    />
    <h5
      className="text-light card-title"
    >
       
      -
       
    </h5>
    <a
      className="btn btn-warning"
      href="/singleMovie/undefined"
      onClick={[Function]}
    >
      Details
    </a>
  </div>
</div>
`;
